### PR TITLE
[#35] Migrate project to start using testcontainers

### LINKED ISSUE #35: Migrate project to start using testcontainers
### Enhancement Description

Currently, the project relies on [flapdoodle.embed.mongo](https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo). The idea is to migrate this dependency to start using [testcontainers](https://www.testcontainers.org/modules/databases/mongodb).

### Current Behaviour

Dependency on `flapdoodle.embed.mongo`.

### Wanted Behaviour

No dependency on `flapdoodle.embed.mongo`.
Added dependency on `testcontainers.mongodb`.
Refactor [MongoLauncher](https://github.com/AxonFramework/extension-mongo/blob/master/mongo/src/test/java/org/axonframework/extensions/mongo/utils/MongoLauncher.java) to rely on `testcontainers.mongodb`.

### PR DESCRIPTION
**Pull request by smcvb**
This pull request removes the usages of "flap doodle" as the means to run embedded Mongo in favour of using Test Containers.
As such, this PR changes the dependencies and the five tests (Saga, TokenStore and 3 EventStorageEngine tests) to use Mongo Test Containers. 

These five tests used a mix of a custom `MongoLauncher` and Spring to wire everything.
Lastly, the `MongoLauncher` (which used "flap doodle") and the `MongoTestContext` were no longer used through these changes and thus have been removed.

This PR resolves #35

### CODE REVIEW

File: mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java [hunk 1]
```diff
@@ -17,84 +17,63 @@
...
     @BeforeEach
     void setUp() {
-        MongoClient mongoClient = null;
-        try {
-            mongoClient = context.getBean(MongoClient.class);
-        } catch (Exception e) {
-            assumeTrue(true, "No Mongo instance found. Ignoring test.");
-        }
-        mongoTemplate = DefaultMongoTemplate.builder().mongoDatabase(mongoClient).build();
+        MongoSettingsFactory mongoSettingsFactory = new MongoSettingsFactory();
```
Comment (YvonneCeelie) [inline mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java:54]: Is it possible to move this to a different class and reuse this code?

**Review by YvonneCeelie — APPROVED** [review]
LGTM

File: mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java (hunk 1 above)
Comment (smcvb) [reply mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java:54]: Think so. Some form of utility class I guess. Can have a look what that would look like.

Comment (smcvb) [reply mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java]: Made the suggested adjustment. Please let me know what you think!

**Review by lfgcampos — APPROVED** [review]
Very nice and lots of lines removed!! 👏

**Pull request merged**